src/rpdk/core/jsonutils/utils.py

```diff
@@ -138,11 +140,17 @@ def schema_merge(target, src, path):  # noqa: C901
             try:
                 target[key] = schema_merge(target_schema, src_schema, next_path)
             except TypeError:
-                if key == "type":
```

Comment:
want to put this up above but can't on github, but can we add the cases to test this new logic to the doctests? would like it just to visualize what the additions to the method are doing w minimal examples.